expr/filter.go

```diff
@@ -11,25 +11,40 @@ import (
 	"github.com/brimdata/zed/zcode"
 )
 
-type Filter func(*zed.Value) bool
+type Filter func(Context, *zed.Value) bool
 
 func LogicalAnd(left, right Filter) Filter {
-	return func(p *zed.Value) bool { return left(p) && right(p) }
+	return func(ectx Context, val *zed.Value) bool {
```

Comment:
Do we want to call the zed.Value parameter `this` in Filter implementation as we do in Eval implementation?